docs/_tutorials/bert-pretraining.md

```diff
@@ -0,0 +1,319 @@
+---
+title: "Pre-training Bing BERT"
+excerpt: ""
+---
+
+In this tutorial we will apply DeepSpeed to pre-train the BERT
+(**B**idirectional **E**ncoder **R**epresentations from **T**ransformers),
+which is widely used for many Natural Language Processing (NLP) tasks. The
+details of BERT can be found here: [BERT: Pre-training of Deep Bidirectional
+Transformers for Language Understanding](https://arxiv.org/abs/1810.04805).
+
+
+We will go through how to setup the data pipeline and how to run the original
+BERT model. Then we will show step-by-step how to modify the model to
+leverage DeepSpeed. Finally, we demonstrate the performance evaluation and
+memory usage reduction from using DeepSpeed.
+
+## Pre-training Bing BERT without DeepSpeed
+
+We work from a adaptations of
+[huggingface/transformers](https://github.com/huggingface/transformers) and
+NVIDIA. We have forked this repo under
+[DeepSpeedExamples/bing_bert](https://github.com/microsoft/DeepSpeedExamples/tree/master/bing_bert)
+and made several modifications in their script:
+  * We adopted the modeling code from NVIDIA's BERT under `bing_bert/nvidia/`.
+  * We extended the data pipeline from [Project Turing](https://msturing.org/)
+    under `bing_bert/turing/`.
+
+Before using DeepSpeed, it is recommended to first go over the pre-requisites
+from the original model README, such as the overview of the model's structure
+and how to run the model.
+
+### Training Data Setup
+
+Download the Wikipedia and BookCorpus datasets and specify their paths in the
+model config file `DeepSpeedExamples/bing_bert/bert\_large\_adam\_seq128.json`:
+
```

Comment:
Can we add text saying that download/preprocessing instructions are coming soon?

Reply:
Done